### ISSUE TITLE
Python Plugin: `vim.buffers` index counts from 0 rather than 1

### ISSUE THREAD

**Issue by HerringtonDarkholme** [neovim/neovim#2823]
I'm on Mac OSX, with `NVIM 0.0.0-alpha+201506042120` +python.

The following problem is all tested against `vim -u NONE` or `nvim -u NONE`
In command mode, `:py print(vim.buffers[0])` in vim raises a exception that buffers should start at 1.
But in neovim, `vim.buffers[0]` gives the buffer object with `bufnr` 1.

I think this is a compatibility problem. Hope this is not reported before as I cannot search a similar issue.

**Comment by Shougo** [comment]
Oh, it should be fixed.

**Comment by splinterofchaos** [comment]
Is this a problem with neovim's API or the python client?

```
`nvim.buffers[0]` is translated to session.request('vim_get_buffers')[0].
```
https://github.com/neovim/python-client/blob/master/neovim/api/common.py#L82

Since `buffers[i]` should refer to the buffer numbered `i`, not the `i`th element in the list, maybe it should search for `i` and error if not found.

**Comment by ZyX-I** [comment]
@splinterofchaos 

> Since buffers[i] should refer to the buffer numbered i, not the ith element in the list, maybe it should search for i and error if not found.

This depends on which version of Vim you are using. Prior to my patches in `vim.buffers[i]` `i` referred to position in a list. After my changes were merged `vim.buffers[i]` started to mean “buffer number i” (`i == vim.buffers[i].number`). Main rationale that justified such an incompatible change was that in first case exact numbers were useless and to get needed buffer one would need to use `for buffer in vim.buffers: if buffer.number == i: return buffer` in any case and *this* code is still working. Though there were some plugins that incorrectly used `vim.buffers[bufnr - 1]` I never cared about plugins that use broken code.

(Theoretically one could’ve used `for i in range(len(vim.buffers)): buffer = vim.buffers[i] …` for iterating over buffers and that will break, but it is highly non-Pythonic and would produce variable `i` which has no valid use-cases I can think about.)

**Comment by justinmk** [comment]
Yeah, this is a problem in the python client.

**Comment by justinmk** [comment]
This is definitely a problem in the python client. Should we make it the API behavior as well?

**Comment by justinmk** [comment]
@bfredl Any idea if this was fixed?

**Comment by JD557** [comment]
This seems to be fixed (at least `vim-worksheet` seems to be working fine on neovim now). Can anyone confirm this?

**Comment by jamessan** [comment]
This was fixed by https://github.com/neovim/python-client/commit/8eb6ba706f6c572cb8b809c3c4a18a3018a3ab15 which was released in python-client 0.1.9.
